### PR TITLE
1345: Added test that reproduces link oversubscribe issue

### PR DESCRIPTION
**Pull request by rtretyak**
related to #1345

### CODE REVIEW

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/MultiRerouteSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        }
+        def currentPath = allPaths.min { pathHelper.getCost(it) }
+        List<FlowPayload> flows = []
+        2.times {
+            def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+            flow.maximumBandwidth = 10000
+            flowHelper.addFlow(flow)
+            flows << flow
+        }
+        assert flows.every { pathHelper.convert(northbound.getFlowPath(it.id)) == currentPath }
```
Comment (ylobankov) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/MultiRerouteSpec.groovy]: flows.each { assert ... }

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/MultiRerouteSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        when: "Make another path more preferable"
+        def newPath = allPaths.find { it != currentPath }
+        allPaths.findAll { it != newPath }.each { pathHelper.makePathMorePreferable(newPath, it) }
+
+        and: "Make preferable path's ISL to have not enough bandwidth to handle 2 flows together, but enough for 1 flow"
+        def currentIsls = pathHelper.getInvolvedIsls(currentPath)
+        def newIsls = pathHelper.getInvolvedIsls(newPath)
+        def thinIsl = newIsls.find { !currentIsls.contains(it) }
+        long newBw = flows.sum { it.maximumBandwidth } - 1
+        database.updateLinkMaxBandwidth(thinIsl, newBw)
```
Comment (ylobankov) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/MultiRerouteSpec.groovy]: Looks like can be written shorter:
```
[thinIsl, thinIsl.reversed].each { database.updateLinkMaxBandwidth(it, newBw) }

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/MultiRerouteSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        database.updateLinkMaxBandwidth(thinIsl.reversed, newBw)
+        database.updateLinkAvailableBandwidth(thinIsl, newBw)
```
Comment (ylobankov) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/MultiRerouteSpec.groovy]: ditto.

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/MultiRerouteSpec.groovy [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        database.updateLinkAvailableBandwidth(thinIsl.reversed, newBw)
+
+        and: "Init simultaneous reroute of both flows by bringing current path's ISL down"
+        def islToBreak = currentIsls.find { !newIsls.contains(it) }
+        northbound.portDown(islToBreak.srcSwitch.dpId, islToBreak.srcPort)
+        TimeUnit.SECONDS.sleep(rerouteDelay - 1)
+
+        then: "Both flows change their paths (or go Down if no path)"
+        Wrappers.wait(5) {
```
Comment (ylobankov) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/MultiRerouteSpec.groovy]: Not sure about this magic constant. Probably we can use WAIT_OFFSET here.

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/MultiRerouteSpec.groovy [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+        and: "'Thin' ISL is not oversubscribed"
+        islUtils.getIslInfo(thinIsl).get().availableBandwidth == newBw - flows.first().maximumBandwidth
+
+        and: "Only one flow goes through a preferred path"
+        flows.count { pathHelper.convert(northbound.getFlowPath(it.id)) == newPath } == 1
+
+        and: "Cleanup: revert system to original state"
+        northbound.portUp(islToBreak.srcSwitch.dpId, islToBreak.srcPort)
+        flows.every { flowHelper.deleteFlow(it.id) }
```
Comment (ylobankov) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/MultiRerouteSpec.groovy]: `every` is fine, but `each` is used everywhere in such kind of operations in tests.

**Comment by rtretyak** [comment]
All above fixed, thanks for review

**Pull request merged**